src/services/getErc20Info.ts

```diff
@@ -0,0 +1,30 @@
+import { Erc20Api } from 'types'
+import { log } from 'utils'
+
+interface Params {
+  address: string
+  erc20Api: Erc20Api
+}
+
+interface Erc20Info {
+  symbol: string
+  name?: string
+  decimals: number
+}
+
+/**
+ * Fetches info for an arbitrary ERC20 token from given address
+ */
+export async function getErc20Info({ address: tokenAddress, erc20Api }: Params): Promise<Erc20Info | null> {
+  try {
+    const [symbol, name, decimals] = await Promise.all([
+      erc20Api.symbol({ tokenAddress }),
+      erc20Api.name({ tokenAddress }),
+      erc20Api.decimals({ tokenAddress }),
+    ])
+    return { symbol, name, decimals }
+  } catch (e) {
+    log('Failed to fetch ERC20 details for address %s', tokenAddress, e)
+    return null
```

Comment:
What does `null` mean?
If it is, there's no token in that address, I think it's fine. But for the implementation I would argue that we should allow to get info of tokens with no symbol/name/decimals. 

I suggest doing a trivial check, like verifying it is a smart contract (has code in that address), and querying the balance for example. If those are true, we assume is an Erc20 an we return an object with all fields optional (except the address if you want to include that). 
If it's not an ERC20 we return `null`

Reply:
This goes a little into what I said in another comment. Do we want to list a token that doesn't have `symbol` or `decimals`?

If we do, how can I check if there's an ERC20 compliant token at a given address?

Reply:
One way, is like I commented above. You can do a basic test (u don't need to be exhaustive)
1. Check that it's a contract --> the address has code in it
2. Try to see if one of the ERC20 functions works. Use the one u like the most (approve, allowace, ...) I think chances are that it is a ERC20.

Reply:
Done